parquet-avro/src/main/java/org/apache/parquet/avro/AvroWriteSupport.java

```diff
@@ -425,4 +285,375 @@ private static GenericData getDataModel(Configuration conf) {
+      }
+    }
+
+    @Override
+    protected void writeByteArray(byte[] array) {
+      if (array.length > 0) {
+        recordConsumer.startField(LIST_REPEATED_NAME, 0);
+        recordConsumer.startGroup(); // repeated group array, middle layer
+        recordConsumer.startField(LIST_ELEMENT_NAME, 0);
+        for (byte element : array) {
+          recordConsumer.addInteger(element);
+        }
+        recordConsumer.endField(LIST_ELEMENT_NAME, 0);
+        recordConsumer.endGroup();
+        recordConsumer.endField(LIST_REPEATED_NAME, 0);
+      }
+    }
+
+    @Override
+    protected void writeShortArray(short[] array) {
+      if (array.length > 0) {
+        recordConsumer.startField("array", 0);
+        recordConsumer.startGroup(); // repeated group array, middle layer
+        recordConsumer.startField(LIST_ELEMENT_NAME, 0);
+        for (short element : array) {
+          recordConsumer.addInteger(element);
+        }
+        recordConsumer.endField(LIST_ELEMENT_NAME, 0);
+        recordConsumer.endGroup();
+        recordConsumer.endField("array", 0);
+      }
+    }
+
+    @Override
+    protected void writeCharArray(char[] array) {
+      if (array.length > 0) {
+        recordConsumer.startField("array", 0);
+        recordConsumer.startGroup(); // repeated group array, middle layer
+        recordConsumer.startField(LIST_ELEMENT_NAME, 0);
+        for (char element : array) {
+          recordConsumer.addInteger(element);
+        }
+        recordConsumer.endField(LIST_ELEMENT_NAME, 0);
+        recordConsumer.endGroup();
+        recordConsumer.endField("array", 0);
+      }
+    }
+
+    @Override
+    protected void writeIntArray(int[] array) {
+      if (array.length > 0) {
+        recordConsumer.startField("array", 0);
+        recordConsumer.startGroup(); // repeated group array, middle layer
+        recordConsumer.startField(LIST_ELEMENT_NAME, 0);
+        for (int element : array) {
+          recordConsumer.addInteger(element);
+        }
+        recordConsumer.endField(LIST_ELEMENT_NAME, 0);
+        recordConsumer.endGroup();
+        recordConsumer.endField("array", 0);
+      }
+    }
+
+    @Override
+    protected void writeLongArray(long[] array) {
+      if (array.length > 0) {
+        recordConsumer.startField("array", 0);
+        recordConsumer.startGroup(); // repeated group array, middle layer
+        recordConsumer.startField(LIST_ELEMENT_NAME, 0);
+        for (long element : array) {
+          recordConsumer.addLong(element);
+        }
+        recordConsumer.endField(LIST_ELEMENT_NAME, 0);
+        recordConsumer.endGroup();
+        recordConsumer.endField("array", 0);
+      }
+    }
+
+    @Override
+    protected void writeFloatArray(float[] array) {
+      if (array.length > 0) {
+        recordConsumer.startField("array", 0);
+        recordConsumer.startGroup(); // repeated group array, middle layer
+        recordConsumer.startField(LIST_ELEMENT_NAME, 0);
```

Comment:
Good point, then I can combine all of the type-specific methods on the ListWriter class, too.